CMakeLists.txt

```diff
@@ -186,13 +191,22 @@ if (USE_SMM MATCHES "blas")
   message("-- Using BLAS for Small Matrix Multiplication")
 elseif (USE_SMM MATCHES "libxsmm")
   # rely on pkg-config in order to link against libxsmm
-  pkg_check_modules(deps REQUIRED IMPORTED_TARGET GLOBAL libxsmmf)
+  pkg_check_modules(LIBXSMM REQUIRED IMPORTED_TARGET GLOBAL libxsmmf)
   message("-- Using libxsmm for Small Matrix Multiplication")
 else ()
   message(FATAL_ERROR "Unknown SMM library specified")
 endif ()
 
-# =================================== GPU backend
+# =================================== GPU backends
+if (USE_OPENCL)
+  if (NOT LIBXSMM_FOUND)
+    message(FATAL_ERROR "LIBXSMM is not found but required for ACC/OpenCL.")
```

Comment:
If the requirement for using OpenCL within DBCSR is to use `libxsmm` for the SMM operations, then the message should be: `"The OpenCL acceleration supports requires DBCSR to use libxsmm as the SMM backend, please install libxsmm and specify USE_SMM=xsmm`, if the user can use the OpenCL accel but still use BLAS as the SMM backend then simply repeat the `pkg_check_modules(LIBXSMM ...)` again (CMake should cache the result from earlier and not repeat the call). The other option is to move the `pkg_check_modules(LIBXSMM...)` outside without the `REQUIRED` and then use `if (NOT LIBXSMM_FOUND)` in both cases.

Reply:
Hmm, indeed this is a more helpful error message. In fact, BLAS can be used for host/cpu side SMMs and I was planning to just keep it simply and require LIBXSMM. However, I will try to decouple things.

Reply:
Well, it is not necessary to untangle it, there may be good reasons to use LIBXSMM as SMM backend when you already have it (I doubt that an AMD EPYC together with an Intel Xe becomes a common setup such that you would want to use BLIS+OpenCL).

Reply:
LIBXSMM works well on AMD. We do CPUID-feature based code-gen/dispatch which yields the best-to-our-knowledge code-path. I doubt that BLIS can compete here as it does not focus on small matrix multiplications (SMMs). CPUID is a necessity for JIT'ted code as we need to figure out what code we want to generate. Further, one can force any code path at runtime even code that fails to run on a given arch (`LIBXSMM_TARGET` environment variable). Btw, we included ARM64 support just now (master) and will upstream this with Linux distributions with our next release.

Reply:
For the record, I've already tried libxsmm on AMD and it is better than BLIS...

Reply:
I have implemented `USE_ACCEL=opencl` and `USE_SMM=libxsmm` to be independent.